curvefs/src/mds/topology/topology_manager.cpp

```diff
@@ -0,0 +1,860 @@
+/*
+ *  Copyright (c) 2021 NetEase Inc.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+/*
+ * Project: curve
+ * Created Date: 2021-08-26
+ * Author: wanghai01
+ */
+
+#include "curvefs/src/mds/topology/topology_manager.h"
+
+#include <sys/time.h>
+#include <sys/types.h>
+
+#include <set>
+#include <string>
+#include <list>
+#include <vector>
+#include <chrono>  //NOLINT
+#include <thread>  //NOLINT
+
+#include "brpc/channel.h"
+#include "brpc/controller.h"
+#include "brpc/server.h"
+
+#include "src/common/concurrent/concurrent.h"
+#include "src/common/concurrent/name_lock.h"
+
+namespace curvefs {
+namespace mds {
+namespace topology {
+
+void TopologyManager::Init(const TopologyOption &option) {
+        option_ = option;
+}
+
+void TopologyManager::RegistMetaServer(
+    const MetaServerRegistRequest *request,
+    MetaServerRegistResponse *response) {
```

Comment:
Actually, there is usually only one metaserver. How do you consider about this?

Reply:
> Actually, there is usually only one metaserver on a server. How do you consider about this?

It dosen't matter, if only one metaserver on a server, it just like mds style.